### ISSUE TITLE
[static-components] Fix false positive when external component is wrapped by useMemo

### ISSUE THREAD

**Issue by Rel1cx** [Rel1cx/eslint-react#1762]
## Background
Per `static-components.spec.diff.md`, when an external component is reassigned via a `useMemo` call (e.g. `useMemo(() => Component, [])`), `resolveDynamicValue` returns the `CallExpression` node without inspecting the callback's return value, causing the variable to be incorrectly flagged as dynamically created.

## Current behavior
```jsx
function Component() {
  const Memoized = useMemo(() => Message, []); // ❌ false positive
  return <Memoized />;
}
```

## Expected behavior
In `resolveDynamicValue`, recognize known memoization hooks (`useMemo`, `useCallback`, etc.) and trace the callback's return value instead of treating the call itself as a dynamic source.

## References
- Diff report: `plugins/eslint-plugin-react-x/src/rules/static-components/static-components.spec.diff.md` (§7 / Key Gaps #7)
- SPEC: `ValidateStaticComponents`

## Priority
P1 — medium implementation cost, high benefit (eliminates a common false positive).

**Comment by Ech0xff** [comment]
Is anyone working on this? If not, I'd like to give it a try.
